### PR TITLE
Fix for string formatting error in Repository.py

### PR DESCRIPTION
**Pull request by darryldixon**
Trivial fix for a longstanding string format bug.

### CODE REVIEW

**Comment by icemac** [comment]
@mauritsvanrees Do you know enough of the code of this project to make a review of this PR? If not who else could?

File: src/Products/ZopeVersionControl/Repository.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def replaceState(self, obj, new_state):
         if obj.__class__ is not new_state.__class__:
             raise VersionControlError(
                 "The class of the versioned object has changed. %s != %s"
-                % (repr(obj.__class__, new_state.__class__)))
+                % (repr(obj.__class__), repr(new_state.__class__)))
```
Comment (ale-rt) [inline src/Products/ZopeVersionControl/Repository.py:79]: IMHO better:
```python
"The class of the versioned object has changed. %r != %r" % (obj.__class__, new_state.__class__)
```
but the current version is good as well

File: src/Products/ZopeVersionControl/Repository.py [hunk 2]
```diff
@@ -100,7 +100,11 @@ def isAVersionableResource(self, obj):
 
     security.declarePublic('isUnderVersionControl')
     def isUnderVersionControl(self, object):
-        return hasattr(object, '__vc_info__')
+        info = getattr(object, '__vc_info__', None)
+        if info is None:
+            return False
+        return info.history_id in self._histories and \
```
Comment (ale-rt) [inline src/Products/ZopeVersionControl/Repository.py:106]: IMHO better and cleaner:
```python
history = self._histories.get(info.history_id)
return history and history.hasVersionId(info.version_id)
```
but the current version is good as well

File: src/Products/ZopeVersionControl/nonversioned.py [hunk 3]
```diff
@@ -138,7 +138,10 @@ def getNonVersionedData(self):
                     # subobjects that are references.
                     continue
             contents[name] = aq_base(value)
-        return {'contents': contents, 'attributes': attributes}
+        order = []
+        if getattr(self.obj, '_objects', False):
+            order = [x['id'] for x in self.obj._objects]
```
Comment (ale-rt) [inline src/Products/ZopeVersionControl/nonversioned.py:143]: IMHO better and more coincise:
```python
order = [obj["id"] for obj in getattr(self.obj, '_objects', [])] 
```
but the current version works as well

**Comment by mauritsvanrees** [comment]
Looking at the [contributors](https://github.com/zopefoundation/Products.ZopeVersionControl/graphs/contributors) I have done a fair number of commits, but they are all packaging related.
@hannosch seems to have done the most, but his last commit was from 2013.
@pbauer did some, but that seems only for Python 3 compatibility.

The first commit seems obviously fine, and it matches the title of the PR.
But the other two commits seem to have nothing to do with it. They look okay at a quick glance, but it is not obvious to me what they fix.

File: src/Products/ZopeVersionControl/nonversioned.py [hunk 4]
```diff
@@ -163,3 +166,18 @@ def restoreNonVersionedData(self, data):
                 # a BTreeFolder2, which doesn't need or want the
                 # _objects attribute.
                 # XXX This is a hackish way to check for BTreeFolder2s.
+        # Yes, we're repeating ourselves:
+        if not hasattr(obj, '_tree'):
```
Comment (ale-rt) [inline src/Products/ZopeVersionControl/nonversioned.py:170]: I am missing context and knowledge why this needs to be done if we have no `_tree` :)

**Review by ale-rt — COMMENTED** [review]
I do not feel competent enough to fully eveluate this PR:
- why this is needed?
- which issue it is going to fix?
- will there will be side effects?

Maybe adding a  test will help to clarify things!

All in all it looks sane and seems a nice improvement to me.
I added some suggestions about code style but they can definitely be ignored.

**Comment by darryldixon** [comment]
> The first commit seems obviously fine, and it matches the title of the PR.
> But the other two commits seem to have nothing to do with it. They look okay at a quick glance, but it is not obvious to me what they fix.

I opened the PR after the first commit, and the subsequent commits got added automatically to it, not really sure why. Anyway, the next two commits fix other issues:

Commit 1) When items are imported by ZEXP from another site and have ```__vc_info__``` attributes that don't correspond to anything in the local site's Version History repository, and

Commit 2) When an ordered folder item is cloned, the order information is lost without this fix.

File: src/Products/ZopeVersionControl/nonversioned.py (hunk 4 above)
Comment (mauritsvanrees) [reply src/Products/ZopeVersionControl/nonversioned.py:170]: That seems explained in the existing comment above it: when there is no `_tree` attribute, the object is a BTreeFolder which does not care about an `_objects` attribute, so we don't need to do anything.
I don't know enough to know if this is true, but I trust the original comment.

**Comment by mauritsvanrees** [comment]
> I opened the PR after the first commit, and the subsequent commits got added automatically to it, not really sure why.

If you make a PR from a branch, and then you push more commits to the branch, then the PR is automatically updated. That is why they got added.

They could have been done in different pull requests, that could have speeded up merging one or more of them, because now a reviewer needs to understand the whole PR.

> Anyway, the next two commits fix other issues

Thanks for the explanation, that makes it clearer.

Looks good to me. I will merge. Thank you!

**Review by mauritsvanrees — APPROVED** [review]

**Pull request merged**